src/nvim/charset.c

```diff
@@ -176,7 +176,7 @@ int buf_init_chartab(buf_T *buf, int global)
         ++p;
 
         if (ascii_isdigit(*p)) {
-          c2 = getdigits_int((char_u **)&p);
+          c2 = getdigits_int((char_u **)&p, true, 0);
```

Comment:
In some cases I'm not really sure we want strict-mode, but did it anyway since it's more conservative.  Can revisit later, or wait for a crash report.